openfl-tutorials/experimental/Workflow_Interface_501_FineTuning_LLAMA2.ipynb

```diff
@@ -0,0 +1,874 @@
+  },
+  {
+   "cell_type": "markdown",
+   "id": "963d6095-4377-43b6-a9da-b2d89919f5bf",
+   "metadata": {},
+   "source": [
+    "Welcome to the introductory tutorial for the OpenFL Experimental Workflow Interface! This notebook is designed to guide you through the process of setting up your first horizontal federated learning workflow using the pre-trained Large Language Model Llama-2. The primary objectives of this tutorial are:\r\n",
+    "- Simplify the federated workflow representation.\r\n",
+    "- Help users better understand the steps in federated learning.\r\n",
+    "- Offer a practical introduction to Large Language Models (LLMs), specifically LLAMA-2, and demonstrate how to fine-tune these models using a dataset.\r\n",
+    "\r\n",
+    "This tutorial serves as a basic example, and users are encouraged to adapt and expand upon it to suit their specific needs and requirements.\r\n",
+    "\r\n",
+    "Please note the following prerequisites and considerations for this tutorial:\r\n",
+    "\r\n",
+    "- Prior to starting this tutorial, you must [register and request access to the Meta model](https://ai.meta.com/resources/models-and-libraries/llama-downloads/).\r\n",
+    "- This tutorial utilizes the lvwerra and some Hugging Face libraries.\r\n",
+    "- For information on system requirements and installation steps, please refer to the [Installation Guide](https://huggingface.co/meta-llama).\r\n",
+    "\r\n",
+    "This tutorial is designed to be a stepping stone in your journey with federated learning and Large Language Models. We hope it provides a solid foundation for you to build upon and adapt to your specific needs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "39c3d86a",
+   "metadata": {},
+   "source": [
+    "# What is it?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a7989e72",
+   "metadata": {},
+   "source": [
+    "The workflow interface is a new way of composing federated learning experiments with OpenFL. It was borne through conversations with researchers and existing users who had novel use cases that didn't quite fit the standard horizontal federated learning paradigm. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fc8e35da",
+   "metadata": {},
+   "source": [
+    "# Getting Started"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4dbb89b6",
+   "metadata": {},
+   "source": [
+    "First, we start by installing the necessary dependencies for the workflow interface"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f7f98600",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install git+https://github.com/intel/openfl.git\n",
+    "!pip install -r requirements_workflow_interface.txt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f8574960-8e1d-4a22-a43a-1c8ea4f8ddb9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install -r requirements_workflow_interface.txt"
```

Comment:
duplicate command 
`!pip install -r requirements_workflow_interface.txt` already in previous cell

Reply:
Fixed